sentry/src/main/java/io/sentry/ISpan.java

```diff
@@ -5,6 +5,22 @@
 
 /** Represents performance monitoring Span. */
 public interface ISpan {
+  String NAME_TAG = "sentry-name";
+
+  /**
+   * Sets the span name.
+   *
+   * @param name - span name
+   */
+  void setName(@Nullable String name);
```

Comment:
I propose we don't have these method on Span since this is a Transaction thing only.

The idea of a Transaction to get a name is either through `startTransaction(name)` in which case whatever object instance it creates knows that it's in fact a Transaction (i.e when `finish` is called on that object, a wire-protocol SentryTransaction is created and values copied into it, including the name).

For the edge case of setting a name after the fact (or changing the name) we could have a special-cased tag called: `sentry-transaction-name` and if that's set we'll use that as the name instead of whatever was given on `startTransaction` as a way to allow overriding the name after the fact.

Reply:
is there an use case to make the name mutable? I'd make it immutable and only taking on startTransaction

Reply:
In web frameworks, route may not be known when the transaction is started as the components that resolve the route from HTTP request have not run yet.